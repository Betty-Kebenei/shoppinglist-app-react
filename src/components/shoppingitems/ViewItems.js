import React from 'react';
import _ from 'lodash';
import UpdateItemForm from './UpdateItemForm';

const ViewItems = (props) => {
    const {
        id,
        listname,
        allShoppingItems,
        deleteOneShoppingItem,
        getAllShoppingItems,
        getErrorMessage,
        searchErrorMessage
    } = props;

    // Display shopping items in a table.
    const renderShoppingitems = () => (
        
        _.map(allShoppingItems, item => {
            return (
                <tr key={item.item_id}>
                    <td>{item.itemname}</td>
                    <td>{item.quantity}</td>
                    <td>{item.units}</td>
                    <td>{item.price}</td>
                    <td>{item.currency}</td>
                    <td>

                    <UpdateItemForm
                        props={props}
                        listId={id}
                        listName={listname}
                        itemId={item.item_id}
                        itemName={item.itemname}
                        quantity={item.quantity}
                        units={item.units}
                        price={item.price}
                        currency={item.currency}
                        getAllShoppingItems={getAllShoppingItems}
                    />

                    <button 
                        type="button" 
                        className="btn glyphicon glyphicon-trash text-primary" 
                        onClick={()=>deleteOneShoppingItem(id, item.item_id)}
                        data-toggle="tooltip" 
                        data-placement="top" 
                        title="Delete_item" 
                        />  

                    </td>
                </tr>
            );
        })
    )
    return( 
        <div className="View col-sm-12">
            <h2>Shopping items for: {listname}</h2><br />
            <div className="col-sm-12">
                <table className="table bordered">
                    <thead>
                        <tr>
                            <td>ITEMNAME</td>
                            <td>QUANTITY</td>
                            <td>UNITS</td>
                            <td>PRICE</td>
                            <td>CURRENCY</td>
                            <td>ACTIONS</td> 
                        </tr>
                    </thead>
                    <tbody>
                        {renderShoppingitems()}
                    </tbody>
                </table>
                {getErrorMessage}
                {searchErrorMessage}
            </div>
        </div>
    );
}
export default ViewItems;