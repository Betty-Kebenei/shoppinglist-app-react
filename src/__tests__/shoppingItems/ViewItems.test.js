import React from 'react';
import { shallow, mount, render } from 'enzyme';

import Config from '../../Config';

import ViewItems from '../../components/shoppingitems/ViewItems';

describe('interface of ViewItems component', () => {
    it('should render', () => {
        const wrapper = shallow(<ViewItems />);
        expect(wrapper).toMatchSnapshot();
    });
});