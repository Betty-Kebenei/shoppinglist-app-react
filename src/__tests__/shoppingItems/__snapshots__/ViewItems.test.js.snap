// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`interface of ViewItems component should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ViewItems />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Shopping items for: 
        </h2>,
        <br />,
        <div
          className="col-sm-12"
        >
          <table
            className="table bordered"
          >
            <thead>
              <tr>
                <td>
                  ITEMNAME
                </td>
                <td>
                  QUANTITY
                </td>
                <td>
                  UNITS
                </td>
                <td>
                  PRICE
                </td>
                <td>
                  CURRENCY
                </td>
                <td>
                  ACTIONS
                </td>
              </tr>
            </thead>
            <tbody />
          </table>
        </div>,
      ],
      "className": "View col-sm-12",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Shopping items for: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Shopping items for: ",
          undefined,
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <table
              className="table bordered"
            >
              <thead>
                <tr>
                  <td>
                    ITEMNAME
                  </td>
                  <td>
                    QUANTITY
                  </td>
                  <td>
                    UNITS
                  </td>
                  <td>
                    PRICE
                  </td>
                  <td>
                    CURRENCY
                  </td>
                  <td>
                    ACTIONS
                  </td>
                </tr>
              </thead>
              <tbody />
            </table>,
            undefined,
            undefined,
          ],
          "className": "col-sm-12",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <thead>
                  <tr>
                    <td>
                      ITEMNAME
                    </td>
                    <td>
                      QUANTITY
                    </td>
                    <td>
                      UNITS
                    </td>
                    <td>
                      PRICE
                    </td>
                    <td>
                      CURRENCY
                    </td>
                    <td>
                      ACTIONS
                    </td>
                  </tr>
                </thead>,
                <tbody />,
              ],
              "className": "table bordered",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <tr>
                    <td>
                      ITEMNAME
                    </td>
                    <td>
                      QUANTITY
                    </td>
                    <td>
                      UNITS
                    </td>
                    <td>
                      PRICE
                    </td>
                    <td>
                      CURRENCY
                    </td>
                    <td>
                      ACTIONS
                    </td>
                  </tr>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <td>
                        ITEMNAME
                      </td>,
                      <td>
                        QUANTITY
                      </td>,
                      <td>
                        UNITS
                      </td>,
                      <td>
                        PRICE
                      </td>,
                      <td>
                        CURRENCY
                      </td>,
                      <td>
                        ACTIONS
                      </td>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "ITEMNAME",
                      },
                      "ref": null,
                      "rendered": "ITEMNAME",
                      "type": "td",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "QUANTITY",
                      },
                      "ref": null,
                      "rendered": "QUANTITY",
                      "type": "td",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "UNITS",
                      },
                      "ref": null,
                      "rendered": "UNITS",
                      "type": "td",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "PRICE",
                      },
                      "ref": null,
                      "rendered": "PRICE",
                      "type": "td",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "CURRENCY",
                      },
                      "ref": null,
                      "rendered": "CURRENCY",
                      "type": "td",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "ACTIONS",
                      },
                      "ref": null,
                      "rendered": "ACTIONS",
                      "type": "td",
                    },
                  ],
                  "type": "tr",
                },
                "type": "thead",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [],
                },
                "ref": null,
                "rendered": Array [],
                "type": "tbody",
              },
            ],
            "type": "table",
          },
          undefined,
          undefined,
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Shopping items for: 
          </h2>,
          <br />,
          <div
            className="col-sm-12"
          >
            <table
              className="table bordered"
            >
              <thead>
                <tr>
                  <td>
                    ITEMNAME
                  </td>
                  <td>
                    QUANTITY
                  </td>
                  <td>
                    UNITS
                  </td>
                  <td>
                    PRICE
                  </td>
                  <td>
                    CURRENCY
                  </td>
                  <td>
                    ACTIONS
                  </td>
                </tr>
              </thead>
              <tbody />
            </table>
          </div>,
        ],
        "className": "View col-sm-12",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Shopping items for: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Shopping items for: ",
            undefined,
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <table
                className="table bordered"
              >
                <thead>
                  <tr>
                    <td>
                      ITEMNAME
                    </td>
                    <td>
                      QUANTITY
                    </td>
                    <td>
                      UNITS
                    </td>
                    <td>
                      PRICE
                    </td>
                    <td>
                      CURRENCY
                    </td>
                    <td>
                      ACTIONS
                    </td>
                  </tr>
                </thead>
                <tbody />
              </table>,
              undefined,
              undefined,
            ],
            "className": "col-sm-12",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <thead>
                    <tr>
                      <td>
                        ITEMNAME
                      </td>
                      <td>
                        QUANTITY
                      </td>
                      <td>
                        UNITS
                      </td>
                      <td>
                        PRICE
                      </td>
                      <td>
                        CURRENCY
                      </td>
                      <td>
                        ACTIONS
                      </td>
                    </tr>
                  </thead>,
                  <tbody />,
                ],
                "className": "table bordered",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <tr>
                      <td>
                        ITEMNAME
                      </td>
                      <td>
                        QUANTITY
                      </td>
                      <td>
                        UNITS
                      </td>
                      <td>
                        PRICE
                      </td>
                      <td>
                        CURRENCY
                      </td>
                      <td>
                        ACTIONS
                      </td>
                    </tr>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <td>
                          ITEMNAME
                        </td>,
                        <td>
                          QUANTITY
                        </td>,
                        <td>
                          UNITS
                        </td>,
                        <td>
                          PRICE
                        </td>,
                        <td>
                          CURRENCY
                        </td>,
                        <td>
                          ACTIONS
                        </td>,
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "ITEMNAME",
                        },
                        "ref": null,
                        "rendered": "ITEMNAME",
                        "type": "td",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "QUANTITY",
                        },
                        "ref": null,
                        "rendered": "QUANTITY",
                        "type": "td",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "UNITS",
                        },
                        "ref": null,
                        "rendered": "UNITS",
                        "type": "td",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "PRICE",
                        },
                        "ref": null,
                        "rendered": "PRICE",
                        "type": "td",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "CURRENCY",
                        },
                        "ref": null,
                        "rendered": "CURRENCY",
                        "type": "td",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "ACTIONS",
                        },
                        "ref": null,
                        "rendered": "ACTIONS",
                        "type": "td",
                      },
                    ],
                    "type": "tr",
                  },
                  "type": "thead",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [],
                  },
                  "ref": null,
                  "rendered": Array [],
                  "type": "tbody",
                },
              ],
              "type": "table",
            },
            undefined,
            undefined,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
